File: 2020/6/구명 보트.py
def solution(people, limit):
    answer, start_idx, end_idx = 0, 0, len(people)-1
    people.sort()
    while start_idx<=end_idx:
        temp_sum = people[end_idx]
        if people and temp_sum + people[start_idx] <= limit:
            temp_sum += people[start_idx]
            start_idx+=1
        end_idx-=1
        answer+=1
    return answer

File: 2020/11/구명 보트.java
import java.util.*;

class Solution {
    public int solution(int[] people, int limit) {
        Arrays.sort(people);
        int end = people.length -1;
        int start = 0;
        int answer = 0;
        while(end >= start){
            if(people[start] + people[end] <= limit){
                start++;
            }
            end--;
            answer += 1; 
        }
        return answer;
    }
}